### PR TITLE
qa: make teardown behavior consistent for kclient/fuse mounts

### PR DESCRIPTION
**Pull request by batrick**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

File: qa/tasks/cephfs/fuse_mount.py [hunk 1]
```diff
@@ -418,8 +418,6 @@ def teardown(self):
         """
         super(FuseMount, self).teardown()
 
-        self.umount()
-
         if self.fuse_daemon and not self.fuse_daemon.finished:
             self.fuse_daemon.stdin.close()
             try:
```
Comment (mchangir) [inline qa/tasks/cephfs/fuse_mount.py:423]: out of academic interest, shouldn't the base class teardown() happen **after** the derived class teardown() ?

Comment (batrick) [reply qa/tasks/cephfs/fuse_mount.py:423]: Yes. We live in a special circle of Python's Hell.

**Review by mchangir — APPROVED** [review]
LGTM

**Review by vshankar — APPROVED** [review]
We should backport this I think, yes? @batrick

**Comment by batrick** [comment]
> We should backport this I think, yes? @batrick

I didn't think it necessarily needed backport. There shouldn't be a functional change.

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65530](https://tracker.ceph.com/issues/65530).

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65562](https://tracker.ceph.com/issues/65562).

**Review by joscollin — APPROVED** [review]

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65596](https://tracker.ceph.com/issues/65596).

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Main#2024-04-20

**Pull request merged**